containers/daap-landing-to-raw/src/var/task/landing_to_raw.py

```diff
@@ -28,6 +46,40 @@ def handler(event, context):
     logger.info(f"Destination bucket: {raw_data_bucket}")
     logger.info(f"Destination key: {destination_key}")
 
+    schema_path = config.data_product_config.schema_path(table_name=config.element.name)
+
+    try:
+        registered_schema = read_json_from_s3(schema_path.uri)
+    except botocore.exceptions.ClientError as e:
+        if e.response["Error"]["Code"] == "NoSuchKey":
+            registered_schema = None
+        else:
+            raise
+
+    if registered_schema:
+        registered_schema_columns = extract_columns_from_schema(registered_schema)
+        inferred_schema = infer_glue_schema_from_raw_csv(
+            file_path=BucketPath(key=file_key, bucket=bucket_name),
+            data_product_element=config.element,
+            logger=logger,
+        ).metadata
+        inferred_columns = extract_columns_from_schema(inferred_schema)
+
+        try:
+            validate_data_against_schema(
+                inferred_columns=inferred_columns,
+                registered_schema_columns=registered_schema_columns,
+            )
+        except DataInvalid:
+            logger.error(f"{file_key} invalid; moving to fail location", exc_info=True)
+            s3.copy(
+                CopySource=copy_source,
+                Bucket=raw_data_bucket,
+                Key=fail_key,
+                ExtraArgs=s3_security_opts,
+            )
+            return
+
     s3.copy(
```

Comment:
Priya mentioned about deleting file from land after copy, which we could add in to this PR?